pkg/lightning/backend/local/rate_limiter.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package local
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/time/rate"
+)
+
+const (
+	maxIngestRequestWorkers = 5
+	maxIngestReqPerSec      = 5
+)
+
+// ingestLimiter is used to limit the concurrency of sending ingest requests.
+// It combines two strategies:
+// - limit the max ingest requests in flight.
+// - limit the ingest requests number per second.
+type ingestLimiter struct {
+	ctx     context.Context
+	slots   chan struct{}
+	limiter *rate.Limiter
+}
+
+func newIngestLimiter(ctx context.Context, maxWorkers, maxReqPerSec int) *ingestLimiter {
+	return &ingestLimiter{
+		ctx:     ctx,
+		slots:   make(chan struct{}, maxWorkers),
+		limiter: rate.NewLimiter(rate.Limit(maxReqPerSec), maxReqPerSec),
+	}
+}
+
+func (l *ingestLimiter) Acquire() error {
```

Comment:
mostly we are using `MultiIngest` in recent versions, so there will be multiple SST generated in a single API call, we need to limit by the number of SST

see https://github.com/pingcap/tidb/blob/cc0925eeafcf0925ba6ba54fb4fe1ae179051a52/pkg/lightning/backend/local/region_job.go#L753-L755